web-server/v0.4/components/Comparison/CompareIterations.js

```diff
@@ -0,0 +1,534 @@
+          key: cluster,
+          clusterID: cluster,
+          cluster: clusterLabels[primaryMetric][cluster],
+          primaryMetric: primaryMetric,
+          length: clusterItems
+        })
+      }
+      for (var i = 0; i < maxClusterLength; i++) {
+        graphKeys[primaryMetric].push(i);
+      }
+    }
+    this.setState({tableData: tableData});
+    this.setState({graphKeys: graphKeys});
+    this.setState({clusteredGraphData: clusteredGraphData});
+    this.setState({loading: false});
+  }
+
+  groupBy = (array, cluster, f) => {
+    var groups = {};
+    array.forEach(function(o) {
+      var group = f(o).join('-');
+      groups[group] = groups[group] || [];
+      groups[group].push(o);
+    });
+    var { clusterLabels } = this.state;
+    clusterLabels[cluster] = Object.keys(groups);
+    this.setState({clusterLabels: clusterLabels});
+    return Object.keys(groups).map(function(group) {
+      return groups[group];
+    });
+  }
+
+  generateIterationClusters = (config) => {
+    this.setState({loading: true})
+    this.setState({selectedConfig: config});
+    const { iterations } = this.props;
+    var primaryMetricIterations = [];
+    primaryMetricIterations = _.mapValues(_.groupBy(iterations, 'primary_metric'),
+                            clist => clist.map(iteration => _.omit(iteration, 'primary_metric')));
+    var clusteredIterations = [];                     
+    for (var cluster in primaryMetricIterations) {
+      clusteredIterations = [];
+      if (typeof config == "object" && config.length > 0) {
+        clusteredIterations = this.groupBy(primaryMetricIterations[cluster], cluster, function(item) {
+          var configData = [];
+          for (var filter in config) {
+            configData.push(item[config[filter]]);
+          }
+          return configData;
+        });
+      } else {
+        clusteredIterations = _.mapValues(_.groupBy(primaryMetricIterations[cluster], config),
+                              clist => clist.map(iteration => _.omit(iteration, config)));
+      } 
+      primaryMetricIterations[cluster] = clusteredIterations;
+    }                 
+    this.setState({primaryMetricIterations: primaryMetricIterations});
+    this.retrieveTimeseriesData(primaryMetricIterations);
+    this.parseGraphData(primaryMetricIterations);
+  }
+
+  resetIterationClusters = () => {
+    const { configCategories } = this.props;
+    this.generateIterationClusters(configCategories);
+  }
+
+  renameProp = (oldProp, newProp,{ [oldProp]: old, ...others }) => {
+    return {
+      [newProp]: old,
+      ...others
+    };
+  };
+
+  retrieveTimeseriesData(clusteredIterations) {
+    var iterationRequests = [];
+    for (var primaryMetric in clusteredIterations) {
+      for (var cluster in clusteredIterations[primaryMetric]) {
+        for (var iteration in clusteredIterations[primaryMetric][cluster]) {
+          iterationRequests.push(axios.get('http://pbench.perf.lab.eng.bos.redhat.com/results/' +
```

Comment:
Can we find a way to remove this hard-coded `http://pbench.perf.lab.eng.bos.redhat.com` URL prefix?